### PR TITLE
Add ACCESS-NRI's spack_config to the docker image

### PR DESCRIPTION
**Pull request by harshula**

### CODE REVIEW

File: containers/Dockerfile.base-spack [hunk 1]
```diff
@@ -78,7 +81,15 @@ SHELL ["/bin/bash", "-c"]
 # Install spack
 RUN git clone -c feature.manyFiles=true https://github.com/spack/spack.git ${SPACK_ROOT} --branch ${SPACK_REPO_VERSION} --single-branch --depth=1
 
-RUN sed -i 's/granularity: microarchitectures/granularity: generic/' ${SPACK_ROOT}/etc/spack/defaults/concretizer.yaml
+# Install ACCESS-NRI's spack_config repo
+RUN git clone  https://github.com/ACCESS-NRI/spack_config.git ${SPACK_CONFIG_REPO_ROOT} --branch ${SPACK_CONFIG_REPO_VERSION}
```
Comment (CodeGat) [inline containers/Dockerfile.base-spack:85]:
```suggestion
RUN git clone  https://github.com/ACCESS-NRI/spack_config.git ${SPACK_CONFIG_REPO_ROOT} --branch ${SPACK_CONFIG_REPO_VERSION} --single-branch --depth=1
```

File: containers/Dockerfile.base-spack [hunk 2]
```diff
@@ -78,7 +81,15 @@ SHELL ["/bin/bash", "-c"]
...
+
+RUN ln -s ${SPACK_CONFIG_DIR}/concretizer.yaml ${SPACK_ROOT}/etc/spack/ \
+ && ln -s ${SPACK_CONFIG_DIR}/config.yaml ${SPACK_ROOT}/etc/spack/ \
+ && ln -s ${SPACK_CONFIG_DIR}/repos.yaml ${SPACK_ROOT}/etc/spack/
+# We don't want to use, only relevant for Gadi:
+#     ${SPACK_CONFIG_DIR}/packages.yaml
+#     ${SPACK_CONFIG}/linux/compilers.yaml
```
Comment (CodeGat) [inline containers/Dockerfile.base-spack:92]: If we don't want to use this commented code as part of the CI pipeline (which doesn't deal with Gadi), should it be here?
On Gadi, how would the un-commented code work?

**Review by CodeGat — CHANGES_REQUESTED** [review]
Hey, just a couple things :)

File: containers/Dockerfile.base-spack (hunk 1 above)
Comment (harshula) [reply containers/Dockerfile.base-spack:85]: In the Docker image, I need to be able to test other branches of spack_config. Remember, it's a dev image for me.

File: containers/Dockerfile.base-spack (hunk 2 above)
Comment (harshula) [reply containers/Dockerfile.base-spack:92]: It's not code, it's a documentation.

File: containers/Dockerfile.base-spack (hunk 1 above)
Comment (CodeGat) [reply containers/Dockerfile.base-spack:85]: Ah, noted. Removed suggestion.

File: containers/Dockerfile.base-spack (hunk 2 above)
Comment (CodeGat) [reply containers/Dockerfile.base-spack:92]: I see, resolved.

**Review by CodeGat — APPROVED** [review]

**Pull request merged**